src/blockrelay/mempool_sync.h

```diff
@@ -0,0 +1,188 @@
+// Copyright (c) 2018-2019 The Bitcoin Unlimited developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_MEMPOOL_SYNC_H
+#define BITCOIN_MEMPOOL_SYNC_H
+
+#include "blockrelay/blockrelay_common.h"
+#include "blockrelay/graphene.h"
+#include "consensus/consensus.h"
+
+#include <chrono>
+
+const uint64_t DEFAULT_MEMPOOL_SYNC_MIN_VERSION_SUPPORTED = 0;
+const uint64_t DEFAULT_MEMPOOL_SYNC_MAX_VERSION_SUPPORTED = 0;
+// arbitrary entropy passed to CGrapheneSet an used for IBLT
+const uint32_t IBLT_ENTROPY = 13;
+// any value greater than 2 will use SipHash
+const uint64_t SHORT_ID_VERSION = 2;
+// frequency of synchronization (per peer) in microseconds
+const int64_t MEMPOOLSYNC_FREQ_US = 30 * 1e6;
+const int64_t MEMPOOLSYNC_FREQ_GRACE_US = 5 * 1e6;
+// Use CVariableFastFilter if true, otherwise use CBloomFilter
+const bool COMPUTE_OPTIMIZED = true;
+
+class CMempoolSyncState
+{
+public:
+    std::chrono::time_point<std::chrono::high_resolution_clock> lastUpdated;
```

Comment:
Use https://en.cppreference.com/w/cpp/chrono/steady_clock for all interval timing.  Use the others for recording and display of wall clock time only.  Otherwise the program can get screwed up when the wall clock is changed either manually or during daylight savings time.  Its ok to use std::chrono, especially if you envision this as being useful outside of bitcoind.  But if you want to be similar to the rest of the code and cut through all the namespaces, templates, and over-typing of std then just use the functions in utiltime.h:

/** Returns a monotonically increasing time for interval measurement (in nSec).  This number is unrelated to calendar time and is not affected by mock time during test */
uint64_t GetStopwatch();
/** Returns a monotonically increasing time for interval measurement (in uSec).  This number is unrelated to calendar time and is not affected by mock time during test */
inline uint64_t GetStopwatchMicros() { return GetStopwatch() / 1000; }

Reply:
Went with GetStopwatchMicros.